test/integration/specs/kbaseCellToolbarMenu_test.js

```diff
@@ -0,0 +1,245 @@
+    });
+
+    // Third cell is minimized.
+
+    it('moves a minimized selected second cell up', async () => {
+        await testCellMovement({
+            caseLabel: 'TEST_CASE_3',
+            direction: 'up',
+            selectCell: false
+        });
+    });
+
+    it('moves a minimized selected second cell down', async () => {
+        await testCellMovement({
+            caseLabel: 'TEST_CASE_3',
+            direction: 'down',
+            selectCell: false
+        });
+    });
+
+    // ... so select it and try again.
+
+    it('moves a minimized selected second cell up', async () => {
+        await testCellMovement({
+            caseLabel: 'TEST_CASE_3',
+            direction: 'up',
+            selectCell: true
+        });
+    });
+
+    it('moves a minimized selected second cell down', async () => {
+        await testCellMovement({
+            caseLabel: 'TEST_CASE_3',
+            direction: 'down',
+            selectCell: true
+        });
+    });
+
+    // Fourth cell is expanded
+
+    it('moves an expanded selected fourth cell up', async () => {
+        await testCellMovement({
+            caseLabel: 'TEST_CASE_4',
+            direction: 'up',
+            selectCell: false
+        });
+    });
+
+    it('moves an expanded unselected fourth cell down', async () => {
+        await testCellMovement({
+            caseLabel: 'TEST_CASE_4',
+            direction: 'down',
+            selectCell: false
+        });
+    });
+
+    // ... so select it and try again.
+
+    it('selects and moves an expanded unselected second cell up', async () => {
+        await testCellMovement({
+            caseLabel: 'TEST_CASE_4',
+            direction: 'up',
+            selectCell: true
+        });
+    });
+
+    it('selects and moves an expanded selected second cell down', async () => {
+        await testCellMovement({
+            caseLabel: 'TEST_CASE_4',
+            direction: 'down',
+            selectCell: true
+        });
+    });
+
+    // Fifth cell is expanded
+
+    it('moves an expanded selected fourth cell up', async () => {
+        await testCellMovement({
+            caseLabel: 'TEST_CASE_5',
+            direction: 'up',
+            selectCell: false
+        });
+    });
+
+    it('moves an expanded selected fourth cell down', async () => {
+        await testCellMovement({
+            caseLabel: 'TEST_CASE_5',
+            direction: 'down',
+            selectCell: false
+        });
+    });
+
+    // ... so select it and try again.
+
+    it('selects and moves an expanded last cell up', async () => {
+        await testCellMovement({
+            caseLabel: 'TEST_CASE_5',
+            direction: 'up',
+            selectCell: true
+        });
+    });
+
+    it('selects and moves an expanded last cell down', async () => {
+        await testCellMovement({
+            caseLabel: 'TEST_CASE_5',
+            direction: 'down',
+            selectCell: true
+        });
+    });
+});
```

Comment:
You have already done some lovely refactoring, and the tests in this file are prime candidates for further refactoring. You are consistently altering three dimensions (caseLabel, direction, selectCell), so they can go into a set of loops, and you can generate the text for the `it('...')` function from the variables. I would add a brief description to each test case JSON object, i.e.:

```json
        "TEST_CASE_1": {
             "cellIndex": 1,
             "title": "Narrative Cell Toolbar Testing",
             "description": "minimised first cell",        
             ...
        },
        ...
        "TEST_CASE_5": {
             "cellIndex": 5,
             "body": "Cell 4",
             "description": "expanded last cell",        
             ...
        },
```

Then you can do something like:
```js
const directions = ['up', 'down'];
const doSelectCell = [true, false];

// iterate through the test cases
// testCases is an object with keys TEST_CASE_x and values the test case data
Object.keys(testCases).forEach((testCaseId) => {
  directions.forEach((direction) => {
    doSelectCell.forEach((selectCell) => {
      let testDescription = `moves the ${testCases[testCaseId].description} ${direction}`;
      if (selectCell) {
        testDescription = `selects and ${testDescription}`;
      }
      it(testDescription, async() => {
        await testCellMovement({
          case: testCaseId,
          direction: direction,
          selectCell: selectCell
        });
      });
   });
});

Reply:
(N.b. There are copy-pasta errors in the descriptions for the tests, which is one reason why I would very strongly recommend using nested loops to parameterise and generate your tests.)